src/matrix/kaldi-matrix.cc

```diff
@@ -2717,6 +2717,45 @@ void MatrixBase<Real>::AddToRows(Real alpha, Real *const *dst) const {
   }
 }
 
+template<typename Real>
+void MatrixBase<Real>::ParametricReLU(const MatrixBase<Real> &src, const VectorBase<Real> &a, const VectorBase<Real> &b) {
+  KALDI_ASSERT(SameDim(*this, src));
+  MatrixIndexT num_rows = num_rows_, num_cols = num_cols_, stride = stride_, src_stride = src.stride_;
+  KALDI_ASSERT(a.Dim() == num_cols && b.Dim() == num_cols);
+  Real *data = data_;
+  const Real *src_data = src.data_;
+  const Real *adata = a.Data();
+  const Real *bdata = b.Data();
+   for (MatrixIndexT r = 0; r < num_rows; r++) {
+    for (MatrixIndexT c = 0; c < num_cols; c++)
+      data[c] = (src_data[c] > 0 ? adata[c] * src_data[c] : bdata[c] * src_data[c]);
+    data += stride;
+    src_data += src_stride;
+  }
+
+}
+
+
+template<typename Real>
+void MatrixBase<Real>::DiffParametricReLU(const MatrixBase<Real> &value,
```

Comment:
The name argument 'value' does not say which matrix should be used. It is true that it is consistent with 'DiffSigmoid' and 'DiffTanh', and I saw the comment at the declaration... Okay, let's keep it as it is...